File: backend/guest-service/src/main/scala/com/radovan/scalatra/utils/RegistrationForm.scala
package com.radovan.scalatra.utils

import com.fasterxml.jackson.databind.JsonNode
import com.radovan.scalatra.dto.GuestDto
import jakarta.validation.Valid
import jakarta.validation.constraints.NotNull

import scala.beans.BeanProperty

@SerialVersionUID(1L)
class RegistrationForm extends Serializable{

  @Valid
  @NotNull
  @BeanProperty var user:JsonNode = _

  @Valid
  @NotNull
  @BeanProperty var guest:GuestDto = _
}

File: vue-frontend/src/classes/RegistrationForm.ts
import Guest from "./Guest"
import User from "./User"

export default class RegistrationForm {
  private _user?: User
  private _guest?: Guest

  get user(): User | undefined {
    return this._user
  }

  set user(value: User | undefined) {
    this._user = value
  }

  get guest(): Guest | undefined {
    return this._guest
  }

  set guest(value: Guest | undefined) {
    this._guest = value
  }
}
